internal/docstore/dynamodocstore/dynamo.go

```diff
@@ -455,13 +457,22 @@ func (c *collection) put(ctx context.Context, k driver.ActionKind, doc driver.Do
 		in.ConditionExpression = ce.Condition()
 	}
 	_, err = c.db.PutItemWithContext(ctx, in)
-	if err == nil {
-		if newPartitionKey != "" {
-			_ = doc.SetField(c.partitionKey, newPartitionKey)
+	if err != nil {
+		if ae, ok := err.(awserr.Error); ok && ae.Code() == dyn.ErrCodeConditionalCheckFailedException {
+			if k == driver.Create {
+				return gcerr.Newf(gcerr.AlreadyExists, err, "document already exists")
+			}
+			if rev, _ := doc.GetField(docstore.RevisionField); rev == nil && k == driver.Replace {
+				return gcerr.Newf(gcerr.NotFound, nil, "document not found")
```

Comment:
Should this be `FailedPrecondition`? I feel it is confusing to return "document not found" when I try to put something into a db.

Reply:
But it's a replace. You're effectively saying: "look up the document with this ID, then replace its contents". If there is no such document, then I think NotFound is the right code, just as it would be with a Get.

Reply:
I was talking about the scenario that you change the kind to replace when the action is a put with the revision ID.

Also, are we doing the same thing for update? Also other providers?

I'll leave this up to you.